viperx/bind_env.go

```diff
@@ -0,0 +1,115 @@
+package viperx
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/ory/viper"
+
+	"github.com/ory/x/stringslice"
+)
+
+const (
+	none = iota - 1
+	properties
+	ref
+	allOf
+	anyOf
+	oneOf
+)
+
+var keys = []string{
+	"properties",
+	"$ref",
+	"allOf",
+	"anyOf",
+	"oneOf",
+}
+
+// BindEnvsToSchema uses all keys it can find from ``
+func BindEnvsToSchema(schema json.RawMessage) error {
+	keys, err := getSchemaKeys(string(schema), string(schema), []string{}, []string{})
+	if err != nil {
+		return err
+	}
+	return viper.BindEnv(keys...)
```

Comment:
As I noted previously this usage of BindEnv is incorrect: this function does not accept multiple keys https://github.com/spf13/viper/blob/master/viper.go#L960 It either accepts one parameter (the key) or two parameters (the key and the corresponding ENV_VAR_NAME)

Reply:
Also, how do we deal with the env names? I guess with the effect of `SetEnvKeyReplacer` it will work automatically, so after`viper.BindEnv("foo.bar.baz")` env `FOO_BAR_BAZ` should work, right?

Reply:
Ah, thanks, I overlooked that completely. Fixed and added a test to make sure this works properly.

> Also, how do we deal with the env names? I guess with the effect of SetEnvKeyReplacer it will work automatically, so afterviper.BindEnv("foo.bar.baz") env FOO_BAR_BAZ should work, right?

Exactly!